ministryofjustice/laa-apply-for-criminal-legal-aid
# CRIMAPP-359 Update to govuk-frontend 5.0.0

## Description of change

This PR updates the govuk-frontend library to the most recent version - 5.0.0. It also updates MoJ frontend to 2.0.1 and the govuk-formbuilder gem to 5.0.0 (the most recent versions), to bring them in line with the changes made in govuk-frontend. The gov-components gem was also added. 

Other changes:
* Updated the phase banner to show a different colour and tag for each environment (prod, staging and local)
* Updated the header and navigation to use the components from govuk-components

Change tracker here, which shows all updates made for both breaking changes and non-breaking changes: https://docs.google.com/spreadsheets/d/1KdSpY6bFgI6lg7bMWrkze_4WcC_B08DB3kmUOPxuDlk/edit#gid=0 

All changes in govuk-frontend 5.0.0 are described here: https://github.com/alphagov/govuk-frontend/releases/tag/v5.0.0

## Link to relevant ticket

[CRIMAPP-359](https://dsdmoj.atlassian.net/browse/CRIMAPP-359 )

## Notes for reviewer

## Screenshots of changes (if applicable)

### Before changes:

### After changes:

## How to manually test the feature

Regression test that everything renders properly and looks okay. Check that components using JS work as expected, e.g. the details component. Check that pagination works. [Tag styles ](https://design-system.service.gov.uk/components/tag/) have changed, but everything else should look the same as before. 


[CRIMAPP-359]: https://dsdmoj.atlassian.net/browse/CRIMAPP-359?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
